packages/core/src/pubsub/task-queue.ts

```diff
@@ -33,6 +40,13 @@ export class TaskQueue {
     });
   }
 
+  /**
+   * Add task and wait till it is completed.
+   */
+  run<T>(task: Task<T>): Promise<T> {
+    return this.#pq.add(task)
```

Comment:
Do we need any retry logic slash error handling here like `add()` above has?

Reply:
The point of `run` (as opposed to `add`) is to pass an error to the caller if it is throw inside a task. Note "fire-and-forget" comment for the `doc` method.

So, no special error handling here.

Added that to the doc-comment.

Reply:
Makes sense, thanks for clarifying!